pkg/testsuite/transactions_framework.go

```diff
@@ -342,159 +329,107 @@ func (t *TransactionFramework) DefaultAddress() iotago.Address {
 	}
 }
 
-func WithDelegationEndEpoch(endEpoch iotago.EpochIndex) options.Option[iotago.DelegationOutput] {
-	return func(delegationOutput *iotago.DelegationOutput) {
-		delegationOutput.EndEpoch = endEpoch
+func WithDelegationEndEpoch(endEpoch iotago.EpochIndex) options.Option[builder.DelegationOutputBuilder] {
+	return func(delegationBuilder *builder.DelegationOutputBuilder) {
+		delegationBuilder.EndEpoch(endEpoch)
 	}
 }
 
-func WithDelegationConditions(delegationConditions iotago.DelegationOutputUnlockConditions) options.Option[iotago.DelegationOutput] {
-	return func(delegationOutput *iotago.DelegationOutput) {
-		delegationOutput.Conditions = delegationConditions
+func WithDelegationConditions(delegationConditions iotago.DelegationOutputUnlockConditions) options.Option[builder.DelegationOutputBuilder] {
+	return func(delegationBuilder *builder.DelegationOutputBuilder) {
+		delegationBuilder.Address(delegationConditions.MustSet().Address().Address)
 	}
 }
 
-func WithDelegationAmount(amount iotago.BaseToken) options.Option[iotago.DelegationOutput] {
-	return func(delegationOutput *iotago.DelegationOutput) {
-		delegationOutput.Amount = amount
+func WithDelegationAmount(amount iotago.BaseToken) options.Option[builder.DelegationOutputBuilder] {
+	return func(delegationBuilder *builder.DelegationOutputBuilder) {
+		delegationBuilder.Amount(amount)
 	}
 }
 
 // BlockIssuer options
 
-func WithBlockIssuerFeature(blockIssuerFeature *iotago.BlockIssuerFeature) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		for idx, feature := range accountOutput.Features {
-			if feature.Type() == iotago.FeatureBlockIssuer {
-				accountOutput.Features[idx] = blockIssuerFeature
-				return
-			}
-		}
-
-		accountOutput.Features = append(accountOutput.Features, blockIssuerFeature)
-	}
-}
-
-func AddBlockIssuerKey(key iotago.BlockIssuerKey) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		blockIssuer := accountOutput.FeatureSet().BlockIssuer()
-		if blockIssuer == nil {
-			panic("cannot add block issuer key to account without BlockIssuer feature")
-		}
-		blockIssuer.BlockIssuerKeys = append(blockIssuer.BlockIssuerKeys, key)
-
-		blockIssuer.BlockIssuerKeys.Sort()
+func WithBlockIssuerFeature(keys iotago.BlockIssuerKeys, expirySlot iotago.SlotIndex) options.Option[builder.AccountOutputBuilder] {
+	return func(accountBuilder *builder.AccountOutputBuilder) {
+		accountBuilder.BlockIssuer(keys, expirySlot)
 	}
 }
 
-func WithBlockIssuerKeys(keys iotago.BlockIssuerKeys) options.Option[iotago.AccountOutput] {
-	return func(accountOutput *iotago.AccountOutput) {
-		blockIssuer := accountOutput.FeatureSet().BlockIssuer()
-		if blockIssuer == nil {
-			panic("cannot set block issuer keys to account without BlockIssuer feature")
-		}
-		blockIssuer.BlockIssuerKeys = keys
+func AddBlockIssuerKey(key iotago.BlockIssuerKey) options.Option[builder.AccountOutputBuilder] {
```

Comment:
Shouldn't we use name  WithBlockIssuerKey to keep the convention? Then you need to update WithBlockIssuerKeys. Maybe WithSetBlockIssuerKeys, as both methods differ, becuase socond replace the keys, not add many

Reply:
Maybe the other way around, only this is adding a new element to the original value while others are replacing it. So I'll keep `WithBlockIssuerKeys` and rename `AddBlockIssuerKey` to `WithAddBlockIssuerKey`.